test/sequential/test-child-process-emfile.js

```diff
@@ -1,30 +1,46 @@
 'use strict';
-var common = require('../common');
-var assert = require('assert');
-var spawn = require('child_process').spawn;
-var fs = require('fs');
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+const fs = require('fs');
 
 if (common.isWindows) {
   console.log('1..0 # Skipped: no RLIMIT_NOFILE on Windows');
   return;
 }
 
-var openFds = [];
+const ulimit = Number(child_process.execSync('ulimit -n'));
```

Comment:
P.S. The error message you got on smartos is what I was getting on OS X so maybe check that you weren't trying to set the hard limit lower than the soft limit? (Although it doesn't really matter since we can and, in this case, should set the soft limit as well so we can just use `-n`.)